File: frontend/src/types/vehiculo.ts
export type Condicion = 'nuevo' | 'usado'

export type EstadoVehiculo = 'disponible' | 'reservado' | 'vendido' | 'dado_de_baja'

export interface Imagen {
  id: number
  url: string
}

export interface Vehiculo {
  id: number
  marca: string
  modelo: string
  anio: number
  kilometraje: number
  combustible: string
  transmision: string
  precio: number
  condicion: Condicion
  estado: EstadoVehiculo
  imagenes: Imagen[]
}

export interface ResumenVehiculo {
  id: number
  marca: string
  modelo: string
  anio: number
  precio: number
  condicion: Condicion
  imagen: string
}

export interface PaginaVehiculos {
  datos: ResumenVehiculo[]
  pagina: number
  tamano: number
  total: number
}

export interface VehiculoEntrada {
  marca: string
  modelo: string
  anio: number
  kilometraje: number
  combustible: string
  transmision: string
  precio: number
  condicion: Condicion
  estado: EstadoVehiculo
  imagenes: string[]
}

export interface ResumenVehiculoGestion {
  id: number
  marca: string
  modelo: string
  anio: number
  kilometraje: number
  precio: number
  condicion: Condicion
  estado: EstadoVehiculo
  imagen: string
}

export interface PaginaVehiculosGestion {
  datos: ResumenVehiculoGestion[]
  pagina: number
  tamano: number
  total: number
}

File: backend/internal/models/vehiculo.go
package models

import "time"

// Estado del vehículo en el concesionario.
const (
	EstadoDisponible  = "disponible"
	EstadoReservado   = "reservado"
	EstadoVendido     = "vendido"
	EstadoDadoDeBaja  = "dado_de_baja"
)

// Condicion del vehículo: nuevo o usado.
const (
	CondicionNuevo = "nuevo"
	CondicionUsado = "usado"
)

// Vehiculo es la entidad de GORM que representa una unidad del concesionario.
type Vehiculo struct {
	ID          uint    `gorm:"primaryKey" json:"id"`
	Marca       string  `gorm:"not null" json:"marca"`
	Modelo      string  `gorm:"not null" json:"modelo"`
	Anio        int     `gorm:"not null" json:"anio"`
	Kilometraje int     `json:"kilometraje"`
	Combustible string  `json:"combustible"`
	Transmision string  `json:"transmision"`
	Precio      float64 `gorm:"not null" json:"precio"`
	Condicion   string  `gorm:"not null" json:"condicion"`
	Estado      string  `gorm:"not null;index;default:disponible" json:"estado"`
	Imagenes    []Imagen `json:"imagenes"`
	CreatedAt   time.Time `json:"-"`
	UpdatedAt   time.Time `json:"-"`
}

// TableName define el nombre de la tabla en español.
func (Vehiculo) TableName() string {
	return "vehiculos"
}

// Imagen es una URL asociada a un vehículo (galería del catálogo).
type Imagen struct {
	ID         uint   `gorm:"primaryKey" json:"id"`
	VehiculoID uint   `gorm:"not null;index" json:"-"`
	URL        string `gorm:"not null" json:"url"`
	CreatedAt  time.Time `json:"-"`
}

// TableName define el nombre de la tabla en español.
func (Imagen) TableName() string {
	return "imagenes"
}
